client/core/bond.go

```diff
@@ -257,8 +259,41 @@ func (c *Core) bondStateOfDEX(dc *dexConnection, bondCfg *dexBondCfg) *dexAcctBo
 		}
 	}
 
-	tierDeficit := int64(state.targetTier) - state.tier
-	state.mustPost = tierDeficit + state.weakStrength - state.pendingStrength
+	// This old way is bugged because we have no way of knowing how many bonus
+	// tiers the server has assigned us, because it is not directly reported in
+	// any comms and  to calculate it locally, we would need to know the
+	// server's ban score, which is configurable and not reported. We would also
+	// need to know our own score, but after the ConnectResult, the server
+	// doesn't send score updates.
+	//
+	// The old way
+	// -----------
+	// tierDeficit := int64(state.targetTier) - state.tier
+	// state.mustPost = tierDeficit + state.weakStrength - state.pendingStrength
+	// -----------
+	//
+	// Instead, if the server says we're short, we assume that we have zero
+	// bonus tiers and use the old calculation.
+	serverTierDeficit := int64(state.targetTier) - state.tier + state.weakStrength - state.pendingStrength
+	if serverTierDeficit > 0 {
+		state.mustPost = serverTierDeficit
+	} else {
+		// But if the server says we're tiered up, we may be so because we have
+		// bonus tiers, which ultimately causes us to ignore weak bonds. In
+		// that case, just make sure that our local bond data is sufficient to
+		// get our target tier. For clients with low targetTier and good trading
+		// history, this will be the primary route through which weak bonds are
+		// renewed.
+		// Note: For users with targetTier > 1, unknown bonus tiers can also
+		// cause us to broadcast renewal bonds that are too small, resulting
+		// in split bonds that cost the user unnecessary tx fees. This solution
+		// is only a stop-gap. A proper solution will require more data from the
+		// API so we can know our bonus tiers.
+		bondTierDeficit := int64(state.targetTier) - state.liveStrength - state.pendingStrength + state.weakStrength
```

Comment:
This basically just cancels out the benefit the user gets from good trading history. Their good trading history will not allow them to have less locked in bonds. If we do
``` golang
if state.liveStrength - state.weakStrength + state.pendingStrength == 0 {
    state.mustPost = 1
}
```
then the user posts the minimum they need to maintain the tier they desire.

Reply:
I don't follow. Can you give an example with numbers?

Reply:
targetTier = 2
currentTier = 4
liveStrength = 2
weakStrength = 2
pendingStrength = 0

In this situation, the original logic would post 0, and the updated logic would post 2. However we only need to post one to make sure that we have at least one live bond.

Reply:
If currentTier was 2 there, then would it drop below target for a moment?

Reply:
Yeah. Because if weakStrength would ever bring tier below target tier, the original code will fire. I think one is always enough. It's difficult to reason about though, for some reason.

Reply:
@buck54321 is this what this is referring to?
```
		// Note: For users with targetTier > 1, unknown bonus tiers can also
		// cause us to broadcast renewal bonds that are too small, resulting
		// in split bonds that cost the user unnecessary tx fees.
```
I'm not sure I understand.

Reply:
So in that scenerio, the extra bonds would be needed if our bonus tiers went down? Is this the same extra fees we would incur if we increased our target tier? You mean that they would expire at different times?

Reply:
Yes, they would expire at different times. Where normally we could send a single bond tx with bond weight 4 once every expiry, we would now be sending a weight 1 bond and the user would see their tier drop from 7 to 4 for some unknown reason. To fix it by broadcasting additional bonds would create "parallel tracks", adding more txs per expiry period.

Reply:
I see! Thank you.

@martonp ok with re-posting all weak bonds?

Reply:
> Posting 2 is correct, imo. We should aim to maintain `bonded_tier`, not `effective_tier = bonded_tier + bonus_tiers`. In fact, any attempt to maintain effective tier for low-tier targets would ultimately fail to maintain bonded status, as @JoeGruffins was seeing in #2459.

I think it's not that bad to repost all the bonds and ignore the bonuses.. it's just a few extra bonds being posted, but my understanding was that the reason all orders were being canceled was that there were no bonds at all. If we just ensure that there is at least one bond at all times, wouldn't that be prevented?

Reply:
Oh. We're basing trading limits on bonded tier anyway, so that's the value we have to maintain.
https://github.com/decred/dcrdex/blob/6a7d7132045502faa617a9309cecf0e0fb24c536/server/auth/auth.go#L815